include/sempr/entity/spatial/AbstractPointCloud.hpp

```diff
@@ -0,0 +1,166 @@
+#ifndef SEMPR_ENTITY_SPATIAL_ABSTRACTPOINTCLOUD_HPP_
+#define SEMPR_ENTITY_SPATIAL_ABSTRACTPOINTCLOUD_HPP_
+
+#include <vector>
+#include <memory>
+#include <functional>
+#include <cstdint>
+
+
+namespace sempr { namespace entity {
+
+// A list of channel type recommendations. Feel free to use own or additional type identifications in your use-case.
+enum ChannelType  // Note: Strong typed enum classes do not allow an implicit downcast to int.
+{
+    N_X = 0,    // Normale in X                     (double)
+    N_Y = 1,    // Normale in Y                     (double)
+    N_Z = 2,    // Normale in Z                     (double)
+    V = 4,      // Velocity in normal direction     (float)
+    I = 5,      // Intensity, normalized            (float)
+    R = 10,     // Red                              (uint8_t)
+    G = 11,     // Green                            (uint8_t)
+    B = 12,     // Blue                             (uint8_t)
```

Comment:
Yes it could be grouped into one channel.
There is also a random access for the channels which allows it iterate only once to get multiple channel information.